src/Listener/MonitoringTokenListener.php

```diff
@@ -0,0 +1,69 @@
+<?php declare(strict_types=1);
+
+namespace Becklyn\Monitoring\Listener;
+
+use Becklyn\Monitoring\Config\MonitoringConfig;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+
+class MonitoringTokenListener implements EventSubscriberInterface
+{
+    /**
+     * @var MonitoringConfig
+     */
+    private $config;
+
+
+    /**
+     * @param MonitoringConfig $config
+     */
+    public function __construct (MonitoringConfig $config)
+    {
+        $this->config = $config;
+    }
+
+
+    /**
+     * @param FilterResponseEvent $event
+     */
+    public function onResponse (FilterResponseEvent $event) : void
+    {
+        // skip if not master request
+        if (!$event->isMasterRequest())
+        {
+            return;
+        }
+
+        $response = $event->getResponse();
+
+        // skip if not HTML response
+        if (false === \strpos($response->headers->get("Content-Type"), "text/html"))
```

Comment:
Do we need to exclude error pages?

Reply:
No, because if they don't have content (and we get a 404) we still want to ensure that that page is up.

Reply:
It is the task of the uptime monitoring to correctly filter the status codes.

Reply:
Fair enough.

I guess the only other special case that may come to my mind (and that is mostly used by third-party bundles) would be JSONP. Unless the return type isn't `text/html` (I can't remember correctly). Either way, that case is handled by `strpos`, so I think we're good to go then.